server/graphql/asset/src/types.rs

```diff
@@ -206,3 +210,127 @@ impl AssetSortInput {
         }
     }
 }
+
+// Asset log types
+
+#[derive(Enum, Copy, Clone, PartialEq, Eq)]
+#[graphql(rename_items = "camelCase")]
+pub enum AssetLogSortFieldInput {
+    Status,
+    LogDatetime,
+}
+
+#[derive(InputObject)]
+pub struct AssetLogSortInput {
+    /// Sort query result by `key`
+    key: AssetLogSortFieldInput,
+    /// Sort query result is sorted descending or ascending (if not provided the default is
+    /// ascending)
+    desc: Option<bool>,
+}
+#[derive(InputObject, Clone)]
+
+pub struct AssetLogFilterInput {
+    pub id: Option<EqualFilterStringInput>,
+    pub asset_id: Option<EqualFilterStringInput>,
+    pub status: Option<StringFilterInput>,
+    pub log_datetime: Option<DatetimeFilterInput>,
```

Comment:
Are we planning to let the user enter the time they did the job? Or should we just capture the time it was actioned?

Reply:
Good question - I just assumed, but based on our discussion today I'm guessing that it would be a manual entry